src/utils.ts

```diff
@@ -238,3 +238,33 @@ export function computedAndError<T>(
   const error = computed(() => valueAndError.value[1]);
   return [value, error];
 }
+
+export function getPairCombinations<T>(arr: T[]): [T, T][] {
+  const combinations: [T, T][] = [];
+
+  for (let i = 0; i < arr.length - 1; i++) {
+    for (let j = i + 1; j < arr.length; j++) {
+      combinations.push([arr[i], arr[j]]);
+    }
+  }
+
+  return combinations;
+}
+
+// Calculates euclidean distance for monzos, as it assumes that if points are not of the same dimension, then the missing dimensions have a value of zero.
+export function monzoEuclideanDistance(
```

Comment:
For future reference, this is just `norm(sub(point1, point2))` if we ever get around updating `xen-dev-utils` dependency. For efficiency even taxicab norm suffices in this application.

Reply:
One would also need to remove the equave index right? Otherwise the octave reduction could increase the distance.

Reply:
Yes, but you don't have to figure it out in this PR. Let's focus on a minimum viable product and defer everything we've talked about on Discord to a separate issue.

I suggest that you simply display an error message to the user if the equave isn't 2/1.